src/fireedge/src/modules/components/Forms/Vm/CreateSchedActionForm/fields.js

```diff
@@ -0,0 +1,610 @@
+  }),
+  htmlType: (_, context) => {
+    const values = context?.getValues() || {}
+
+    return (
+      !(
+        values?.PERIODIC === SCHEDULE_TYPE.PERIODIC &&
+        values?.REPEAT === REPEAT_VALUES.WEEKLY
+      ) && INPUT_TYPES.HIDDEN
+    )
+  },
+  validation: lazy((_, { context }) =>
+    array(string())
+      .min(1)
+      .default(() => context?.[DAYS_FIELD.name]?.split?.(',') ?? [])
+      .when(REPEAT_FIELD.name, (repeatType, schema) =>
+        repeatType !== REPEAT_VALUES.WEEKLY
+          ? schema.strip()
+          : schema.required(T.DaysBetween0_6)
+      )
+      .afterSubmit((value) => value?.join?.(','))
+  ),
+  grid: { md: 12 },
+}
+
+/** @type {Field} Monthly field */
+const MONTHLY_FIELD = {
+  name: 'MONTHLY',
+  dependOf: [PERIODIC_FIELD_NAME, REPEAT_FIELD.name],
+  type: INPUT_TYPES.TEXT,
+  label: T.DayOfMonth,
+  htmlType: (_, context) => {
+    const values = context?.getValues() || {}
+
+    return (
+      !(
+        values?.PERIODIC === SCHEDULE_TYPE.PERIODIC &&
+        values?.REPEAT === REPEAT_VALUES.MONTHLY
+      ) && INPUT_TYPES.HIDDEN
+    )
+  },
+  grid: { md: 12 },
+  validation: lazy((_, { context }) =>
+    string()
+      .trim()
+      .matches(MONTH_DAYS_REG, { message: T.DaysBetween1_31 })
+      .default(() => context?.[DAYS_FIELD.name])
+      .when(REPEAT_FIELD.name, (repeatType, schema) =>
+        repeatType !== REPEAT_VALUES.MONTHLY
+          ? schema.strip()
+          : schema.required()
+      )
+  ),
+}
+
+/** @type {Field} Yearly field */
+const YEARLY_FIELD = {
+  name: 'YEARLY',
+  dependOf: [PERIODIC_FIELD_NAME, REPEAT_FIELD.name],
+  type: INPUT_TYPES.TEXT,
+  label: T.DayOfYear,
+  htmlType: (_, context) => {
+    const values = context?.getValues() || {}
+
+    return (
+      !(
+        values?.PERIODIC === SCHEDULE_TYPE.PERIODIC &&
+        values?.REPEAT === REPEAT_VALUES.YEARLY
+      ) && INPUT_TYPES.HIDDEN
+    )
+  },
+  grid: { md: 12 },
+  validation: lazy((_, { context }) =>
+    string()
+      .trim()
+      .matches(YEAR_DAYS_REG, { message: T.DaysBetween0_365 })
+      .default(() => context?.[DAYS_FIELD.name])
+      .when(REPEAT_FIELD.name, (repeatType, schema) =>
+        repeatType !== REPEAT_VALUES.YEARLY ? schema.strip() : schema.required()
+      )
+  ),
+}
+
+/** @type {Field} Hourly field */
+const HOURLY_FIELD = {
+  name: 'HOURLY',
+  dependOf: [PERIODIC_FIELD_NAME, REPEAT_FIELD.name],
+  type: INPUT_TYPES.TEXT,
+  label: T.EachXHours,
+  grid: { md: 12 },
+  htmlType: (_, context) => {
+    const values = context?.getValues() || {}
+
+    return (
+      !(
+        values?.PERIODIC === SCHEDULE_TYPE.PERIODIC &&
+        values?.REPEAT === REPEAT_VALUES.HOURLY
+      ) && INPUT_TYPES.HIDDEN
+    )
+  },
+  validation: lazy((_, { context }) =>
+    number()
+      .min(0)
+      .max(168)
+      .integer()
+      .default(() => context?.[DAYS_FIELD.name])
+      .when(REPEAT_FIELD.name, (repeatType, schema) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).